elastic/kibana
# [ObsUX] Add Missing transaction warning if there are Transactions or Spans with a parent.id that doesn't exist in the trace

Closes https://github.com/elastic/kibana/issues/25117

## Summary

Add a missing transaction warning if there are Transactions or Spans with a `parent.id` that doesn't exist in the trace.


## Testing

- Use the `trace_with_orphan_items.ts` scenario:  `node scripts/synthtrace --clean trace_with_orphan_items.ts`
- In APM -> Traces there are 2 traces:
   - <img width="1423" alt="image" src="https://github.com/elastic/kibana/assets/14139027/3548d1dd-d87f-4090-a028-d62cf8ec35c8">
- Check the traces: 
   - Incomplete trace (with warning):
       - <img width="1401" alt="image" src="https://github.com/elastic/kibana/assets/14139027/a75a2112-a425-43d9-bdd1-b3724ccfe6e3">

   - Complete trace (no warning):
       -  <img width="1387" alt="image" src="https://github.com/elastic/kibana/assets/14139027/f33614ab-269b-447c-bd0b-bc00f9799092">
- Unit test in [waterfall_helpers.test.ts](https://github.com/elastic/kibana/pull/171196/files#diff-6cdeaa931c0085a16353ac34f937d442a39e1227621f11b3de0608a39e949fc6)
